packages/mux-player/src/html.ts

```diff
@@ -0,0 +1,122 @@
+import {
+  TemplateInstance,
+  NodeTemplatePart,
+  createProcessor,
+  AttributeTemplatePart,
+} from "@github/template-parts";
+import type { TemplatePart, TemplateTypeInit } from "@github/template-parts";
+
+function processSubTemplate(part: TemplatePart, value: unknown): boolean {
+  if (value instanceof TemplateResult && part instanceof NodeTemplatePart) {
+    value.renderInto(part);
+    return true;
+  }
+  return false;
+}
+
+function processDocumentFragment(part: TemplatePart, value: unknown): boolean {
+  if (value instanceof DocumentFragment && part instanceof NodeTemplatePart) {
+    if (value.childNodes.length) part.replace(...value.childNodes);
+    return true;
+  }
+  return false;
+}
+
+export function processPropertyIdentity(
+  part: TemplatePart,
+  value: unknown
+): void {
+  part.value = String(value);
+}
+
+export function processBooleanAttribute(
+  part: TemplatePart,
+  value: unknown
+): boolean {
+  if (
+    typeof value === "boolean" &&
+    part instanceof AttributeTemplatePart
+    // can't use this because on custom elements the props are always undefined
+    // typeof part.element[part.attributeName as keyof Element] === 'boolean'
+  ) {
+    part.booleanValue = value;
+    return true;
```

Comment:
non-blocking: We should be able to model these as `is` type predicates (See: https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates)